src/content/docs/new-relic-control/agent-control/architecture.mdx

```diff
@@ -0,0 +1,58 @@
+
+   <img
+     title="Agent Control architecture"
+     alt="Agent Control architecture for Agent Management"
+     src="/images/agent-control-architecture.png"
+   />
+
+
+Agent Control's power comes from its use of a declarative, GitOps-like approach to agent management. It treats the desired state of your instrumentation as a single source of truth, ensuring that what you configure in New Relic Control is what's running on your hosts. This process is driven by key underlying technologies.
+
+### GitOps-like Principles with Flux
+Agent Control leverages Flux, a powerful and widely adopted tool, to manage the state of your agents. This means you define your desired configuration once, and Agent Control automatically ensures that your agents conform to this state. This declarative approach provides several key benefits:
+
+- **Consistency:** Eliminates configuration drift across your hosts and clusters.
+- **Automation:** Automatically applies configuration changes and updates, reducing manual toil.
+- **Reliability:** Provides a self-healing mechanism, as Agent Control continuously monitors and reconciles the state of your agents to match your desired configuration.
+
+### Centralized Remote Configuration
+Agent Control can use local or remote configurations. In the first case, the configurations are read directly, but in the later, they are fetched from New Relic Control and rendered directly to your agents. This allows you to manage everything from a central point (and UI), including:
+
+- **Credentials:** Agent Control can pull sensitive credentials from external secret providers like Vault, rendering them into the agent's configuration at runtime. This enhances security by keeping credentials out of the configuration itself. 
+- **Agent Configuration:** It automatically converts remote configurations into a format that is compatible with the underlying agent. This means you can manage different types of agents (e.g., Infrastructure Agent, OpenTelemetry Collector) with a single, unified process.
+
+### The Future of Agent Management
+
+This powerful architecture lays the foundation for future features that will further enhance your observability experience:
+- **Fleet-wide commands:** The ability to send commands and perform actions on your entire fleet of agents.
+- **Advanced communication:** Direct, real-time communication with agents for on-demand troubleshooting and data collection.
+- **Enhanced security:** Further integration with security providers and automated credential management.
+
+<Callout title="warning">
+    To enable this level of control and communication, Agent Control requires specific permissions. It needs to be configured with the necessary access to pull configurations, and in some cases, interact with local systems to manage the agent lifecycle. When deploying Agent Control, ensure you understand and grant the required permissions to maintain a secure and functional environment.
+</Callout>
```

Comment:
I think the warning is great, but I'm not sure if we could include a link to the section where such required permisions are described.